packages/babel-parser/src/tokenizer/index.js

```diff
@@ -713,12 +717,37 @@ export default class Tokenizer extends ParserErrors {
       // it can be merged with tt.assign.
       this.finishOp(tt.xorAssign, 2);
     }
+    // '^^'
+    else if (hackPipeWithDoubleCaretIsActive && next === charCodes.caret) {
```

Comment:
we can extend `expectPlugin` to accept an additional option check. So if the plugin is not enabled or the plugin option is incorrect, we can provide a better error message:

> This syntax requires experimental syntax plugin "pipelineOperator"

or

> This syntax requires "pipelineOperator" plugin options: { proposal: "hack", topicToken: "@@" }, instead got ....

Reply:
Should we extend `expectPlugin` in this pull request or in a separate one?

Reply:
Since `expectPlugin` is not a public method, extending it can be merged soon, we can open a new PR and rebase this one upon that.

Reply:
#13982 has been merged into master. Having said that, I took a look at this again, and I don’t think we can use `expectPlugin` here, because we are actually tokenizing differently based on the `topicToken`; we are not throwing any errors in the tokenizer based on `topicToken`.